scout/parse/matchmaker.py

```diff
@@ -158,6 +159,18 @@ def genomic_features(store, case_obj, sample_name, candidate_vars, genes_only):
     return g_features
 
 
+def parse_datetime(match_date: Union[str, int]) -> datetime:
+    """Converts a date to a datetime."""
+    try:  # Patientmatcher < 4.5 returns dates in milliseconds
+        date_datetime = datetime.fromtimestamp(match_date / 1000.0)
+    except TypeError:  # Patientmatcher >= 4.5 returns dates in isoformat
+        try:  # try parsing date with fractional seconds
```

Comment:
This looks like it would work well, although any errors are going to be a little hard to trace "another exception occurred during the handling of the exceptions exceptions exception".
ChatGPT suggested checking for a literal '.' 🤖 
```
if '.' in date_string:
    dt = datetime.strptime(date_string, "%Y-%m-%d %H:%M:%S.%f")
else:
    dt = datetime.strptime(date_string, "%Y-%m-%d %H:%M:%S")
```
You could extend that and have the first if be a check for a ms return.. 
But all good, either is fine.